soc/arm/nxp_lpc/lpc55xxx/soc.c

```diff
@@ -58,59 +113,64 @@ const pll_setup_t pll0Setup = {
+	/* Power Management Controller Initialization */
 	POWER_PowerInit();
 #endif
-
-#if defined(CONFIG_SOC_LPC55S06) || defined(CONFIG_SOC_LPC55S16) || \
-	defined(CONFIG_SOC_LPC55S28) || defined(CONFIG_SOC_LPC55S36) || \
-	defined(CONFIG_SOC_LPC55S69_CPU0)
     /*!< Set up the clock sources */
     /*!< Configure FRO192M */
 	/*!< Ensure FRO is on  */
 	POWER_DisablePD(kPDRUNCFG_PD_FRO192M);
 	/*!< Set up FRO to the 12 MHz, just for sure */
-	CLOCK_SetupFROClocking(12000000U);
+	CLOCK_SetupFROClocking(12000000);
 	/*!< Switch to FRO 12MHz first to ensure we can change the clock */
 	CLOCK_AttachClk(kFRO12M_to_MAIN_CLK);
 
-	/* Enable FRO HF(96MHz) output */
-	CLOCK_SetupFROClocking(96000000U);
+#if defined(CONFIG_INIT_PLL0) && !defined(CONFIG_SOC_LPC55S36)
+	/* Enable XTALHF clock */
+	POWER_DisablePD(kPDRUNCFG_PD_XTAL32M);
+	POWER_DisablePD(kPDRUNCFG_PD_LDOXO32M);
+#endif
 
-#ifdef CONFIG_INIT_PLL0
-	/*!< Ensure XTAL16M is on  */
-	PMC->PDRUNCFGCLR0 |= PMC_PDRUNCFG0_PDEN_XTAL32M_MASK;
-	PMC->PDRUNCFGCLR0 |= PMC_PDRUNCFG0_PDEN_LDOXO32M_MASK;
+#if !defined(CONFIG_TRUSTED_EXECUTION_NONSECURE)
+	/* Set Voltage for one of the fastest clock outputs: System clock output */
+	POWER_SetVoltageForFreq(DT_PROP(DT_CLOCKS_CTLR_BY_NAME(SYSCON_NODE, sysclk),
+					clock_frequency));
+	/* Set FLASH wait states for core */
+	CLOCK_SetFLASHAccessCyclesForFreq(DT_PROP(DT_CLOCKS_CTLR_BY_NAME(SYSCON_NODE,
+									 sysclk),
+								clock_frequency));
+#endif
 
-	/*!< Ensure CLK_IN is on  */
+	/*!< Ensure CLK_IN is on is either PLL is enabled  */
+#if ((DT_PROP(DT_CLOCKS_CTLR_BY_NAME(SYSCON_NODE, sysclk), clock_frequency) == 150000000) \
+	|| (defined(CONFIG_INIT_PLL0)))
+	CLOCK_SetupExtClocking(16000000);
+#if !defined(CONFIG_SOC_LPC55S36)
 	SYSCON->CLOCK_CTRL |= SYSCON_CLOCK_CTRL_CLKIN_ENA_MASK;
+#endif
 	ANACTRL->XO32M_CTRL |= ANACTRL_XO32M_CTRL_ENABLE_SYSTEM_CLK_OUT_MASK;
-
-	/*!< Switch PLL0 clock source selector to XTAL16M */
-	CLOCK_AttachClk(kEXT_CLK_to_PLL0);
-
-	/*!< Configure PLL to the desired values */
-	CLOCK_SetPLL0Freq(&pll0Setup);
-
-	CLOCK_SetClkDiv(kCLOCK_DivPll0Clk, 0U, true);
-	CLOCK_SetClkDiv(kCLOCK_DivPll0Clk, 1U, false);
 #endif
 
-#if !defined(CONFIG_TRUSTED_EXECUTION_NONSECURE)
-	/*!< Set FLASH wait states for core */
-	CLOCK_SetFLASHAccessCyclesForFreq(96000000U);
+	/* Configuring the Main Clock output to either 150MHz with PLL1 or 96MHz with FRO */
+#if DT_PROP(DT_CLOCKS_CTLR_BY_NAME(SYSCON_NODE, sysclk), clock_frequency) == 150000000
+	InitPLL1();
+#else
+	CLOCK_SetupFROClocking(96000000);
+	/* Swtich MAIN_CLK to FRO_HF */
+	CLOCK_AttachClk(kFRO_HF_to_MAIN_CLK);
 #endif
 
-    /*!< Set up dividers */
-	CLOCK_SetClkDiv(kCLOCK_DivAhbClk, 1U, false);
+	/* Set AHBCLKDIV divider to value 1 */
+	CLOCK_SetClkDiv(kCLOCK_DivAhbClk, 1, false);
 
-    /*!< Set up clock selectors - Attach clocks to the peripheries */
-	CLOCK_AttachClk(kFRO_HF_to_MAIN_CLK);
+	/* Initialization for PLL0, currently tested for I2S */
+#ifdef CONFIG_INIT_PLL0
```

Comment:
I'm just asking a question here, maybe there is a good reason to use a Kconfig symbol here.  But since we are adding devicetree nodes for each PLL, would it be better to enable the PLL in the DTS with `status = "okay"`, and then initialize each PLL based on the node property?